### ISSUE TITLE
Add eventid customization

### ISSUE THREAD

**Issue by CatinoRosalba** [builttoroam/device_calendar#518]
**Problem**
Creating a new event loaded from a ics file, within it is included a uid which I would like to be the event id. Initially, since I have the ability to assign a value to it, I thought it would actually assign it, but in reality once the event is created, the eventid is overwritten with a causal id.

**Why implement it**
Having a custom eventid and not auto-generated (as it currently is) allows me to perform checks on its existence and be able to determine consequent actions. For example, avoid duplicate events and rather modify it.

**How I am currently solving**
I am solving by creating a local database and then saving the events there, with the custom uid, but it would be nice to be able to manage it through the library itself.

_N.B. In case this possibility already exists and I am doing something wrong, I withdraw my request and thank you for any tip._

**Comment by hamishjohnson** [comment]
This is key for 2 way sync I believe

**Comment by CatinoRosalba** [comment]
Yes, in fact entering the event with an ics file was just the beginning of event synchronization development. Perhaps, however, regardless of synchronization actions, having control over the event id would also allow better management of device events.

**Comment by michalcharvat** [comment]
Any update on this? If would be very useful because guid is generated on the server and in this pub it is read only attribute - you can provide existing but you cannot provide yours.

**Comment by IVLIVS-III** [comment]
Afaik, this is a platform limitation, i.e. the operating system API itself does not allow for a custom eventId, sadly.

You could solve two-way sync by putting a custom id in the notes field. However, keep in mind that users would be able to go into their systems calendar and edit this notes field which could break sync.
